### PR TITLE
[numpy] random.rand

### PR DESCRIPTION
**Pull request by kshitij12345**
## Description ##
Add 
https://docs.scipy.org/doc/numpy-1.16.0/reference/generated/numpy.random.rand.html

Heavily Referenced from #15903 
@xidulu Thank You.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Wrappers
- [x] Test

### CODE REVIEW

File: python/mxnet/numpy/random.py [hunk 1]
```diff
@@ -20,7 +20,36 @@
...
+    Parameters
+    ----------
+    d0, d1, ..., dn : int, optional
+        The dimensions of the returned array, should be all positive.
+        If no argument is given a single Python float is returned.
+    Returns
+    -------
+    out : ndarray
+        A ``(d0, d1, ..., dn)``-shaped array of floating-point samples from
+        the standard normal distribution, or a single such float if
```
Comment (xidulu) [inline python/mxnet/numpy/random.py]: Should remove "from the standard normal".

File: python/mxnet/symbol/numpy/random.py [hunk 2]
```diff
@@ -49,6 +49,35 @@ def _random_helper(random, sampler, params, shape, dtype, ctx, out, kwargs):
...
+    Parameters
+    ----------
+    d0, d1, ..., dn : int, optional
+        The dimensions of the returned array, should be all positive.
+        If no argument is given a single Python float is returned.
+    Returns
+    -------
+    out : ndarray
+        A ``(d0, d1, ..., dn)``-shaped array of floating-point samples from
+        the standard normal distribution, or a single such float if
```
Comment (xidulu) [inline python/mxnet/symbol/numpy/random.py]: Same as above.

File: python/mxnet/numpy/random.py (hunk 1 above)
Comment (kshitij12345) [reply python/mxnet/numpy/random.py]: Will fix it. Thank You.

**Comment by haojin2** [comment]
Please rebase with current master and resolve the conflicts @kshitij12345

**Comment by kshitij12345** [comment]
@haojin2 
Sure will do that and update. 

Rebase with master or numpy branch?

File: python/mxnet/numpy/random.py [hunk 3]
```diff
@@ -20,7 +20,34 @@
...
+    --------
+    >>> np.random.rand(3,2)
+    array([[ 0.14022471,  0.96360618],  #random
+           [ 0.37601032,  0.25528411],  #random
+           [ 0.49313049,  0.94909878]]) #random
+    """
+    output_shape = ()
+    for s in size:
+        output_shape += (s,)
+    return _mx_nd_np.random.uniform(0, 1, size=output_shape, **kwargs)
```
Comment (reminisce) [inline python/mxnet/numpy/random.py]: `uniform` samples over the range `[0, 1]`, while `rand` samples over the range `[0, 1)`. To simulate this behavior using `uniform`, the upper bound should be set to `1 - np.finfo(dtype).eps`.

Comment (reminisce) [reply python/mxnet/numpy/random.py]: I see. Thanks for the contribution.

**Comment by kshitij12345** [comment]
@reminisce @haojin2 @xidulu Sorry for the delayed action. Please review.

**Comment by reminisce** [comment]
@kshitij12345 Thanks for your contribution. It's merged through https://github.com/apache/incubator-mxnet/pull/16554. I added missing implementation in `mxnet/ndarray/numpy/random.py`.

**Comment by kshitij12345** [comment]
@reminisce Oh did not notice it. Thank You. Will close this one.

**Pull request closed without merging**